app/project_type/git.py

```diff
@@ -194,20 +270,24 @@ def _execute_git_remote_command(self, command, cwd=None, timeout=10):
 
                     # Automatically add hosts that aren't in the known_hosts file to the known_hosts file.
                     child.sendline('yes')
-                    self._logger.info('Automatically added a host to known_hosts in command: {}'.format(command))
+                    log_msg_queue.put(
+                        ('INFO', 'Automatically added a host to known_hosts in command: {}', (command,)))
+
             except pexpect.EOF:
                 break
             except pexpect.TIMEOUT:
-                self._logger.info('Command [{}] had no expected prompts after {} seconds.'.format(command, timeout))
+                log_msg_queue.put(
+                    ('INFO', 'Command [{}] had no expected prompts after {} seconds.', (command, timeout)))
                 break
 
         # Dump out the output stream from pexpect just in case there was an unexpected prompt that wasn't caught.
-        self._logger.debug("Output from command [{}] after {} seconds: {}".format(command, timeout, child.before))
+        log_msg_queue.put(
+            ('DEBUG', 'Output from command [{}] after {} seconds: {}', (command, timeout, child.before)))
 
         # Now we assume we are past any prompts and wait for the command to end.  We need to keep checking
         # if the kill event has been set in case the build is canceled during setup.
         finished = None
-        while not self._kill_event.is_set() and finished is None:
+        while finished is None:  # and not self._kill_event.is_set()  # todo: kill_event temporarily disabled
```

Comment:
The kill event was added here with the feature to enable canceling builds. I disabled it while I figure out a clean way to support this across multiple processes (the existing approach would not work). I'm planning on reenabling it very soon. Disabling this shouldn't have much effect in the short term.